starboard/shared/starboard/player/job_thread.cc

```diff
@@ -67,13 +67,20 @@ JobThread::JobThread(const char* thread_name,
 }
 
 JobThread::~JobThread() {
+  if (thread_) {
+    JoinThread();
+  }
+}
+
+void JobThread::JoinThread() {
```

Comment:
We are adding a new convention to `JobThread`, that `JoinThread()` has to be called prior to destruction.

We could handle this situation better by introducing a new class (e.g. `JobThreadHolder`) that holds and JobThread object, keep itself valid before destroying it.  Once the new class is implemented, turn the dtor of JobThread into private and a friend of the new class, so other teammates won't accidentally wrap a JobThread into a std::unique_ptr<>.